### PR TITLE
Fix auth and add heartbeat

### PR DESCRIPTION
**Pull request by tpavs**

### CODE REVIEW

File: events/socket.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+			responsePayload = errorJson
+		} else {
+			responsePayload = *respBytes
+		}
+
+		msg := Message{
+			Name: message.Name,
+			ContentType: message.ContentType,
+			EventId: message.EventId,
```
Comment (lickel) [inline events/socket.go]: This doesn't make a ton of sense to me.
Shouldn't the backend create a new eventID for every message?

File: events/socket.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+			EventAckId: fmt.Sprintf("ack:%s", message.EventId),
```
Comment (lickel) [inline events/socket.go]: eventAckID should === the original eventID

File: events/heartbeat.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+type payload struct {
+	Time uint64 `json:"time"`
+}
+
+type Heartbeat struct {}
+
+const JsonContentType = "application/json"
+
+func (h Heartbeat) apply(message Message) (*[]byte, error) {
+	if message.ContentType != JsonContentType {
```
Comment (lickel) [inline events/heartbeat.go]: Do we need to specify contentType now?

File: events/heartbeat.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+		glog.Errorf("Error unmarshalling payload: %s", jsonError)
+		return nil, &Error{
+			msg: "Malformed frame",
+			code: "badRequest",
+			statusCode: http.StatusBadRequest,
+		}
+	}
+
+	responsePayload := payload{
+		Time: p.Time,
```
Comment (lickel) [inline events/heartbeat.go]: The response payload should be the server's time, not the original payload's time

File: events/heartbeat.go (hunk 3 above)
Comment (tpavs) [reply events/heartbeat.go]: Its in the contract I would rather fail knowingly than implicitly.

File: events/heartbeat.go (hunk 4 above)
Comment (tpavs) [reply events/heartbeat.go]: Wasn't sure and wanted validate end 2 end easy to change will do. 👍

File: events/socket.go (hunk 1 above)
Comment (tpavs) [reply events/socket.go]: ok So client calls w/ 
eventId and we response w/
new eventId and an the ackId of the original event?

Y not just have eventId be constant and the response id goes in ack?

File: events/heartbeat.go (hunk 3 above)
Comment (lickel) [reply events/heartbeat.go]: sure, but I thought contentType's default was json?

File: events/socket.go (hunk 1 above)
Comment (lickel) [reply events/socket.go]: arguably we don't need eventAckId then?

File: events/heartbeat.go (hunk 3 above)
Comment (tpavs) [reply events/heartbeat.go]: I can do that.

File: events/socket.go (hunk 1 above)
Comment (thatbroc) [reply events/socket.go]: Originator of event should be creating unique id's on either side.

**Pull request merged**